experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -781,34 +796,40 @@ export async function generate(
 
         await processTemplates(schema, templateDirs, allLocales, outPath, scope, force, feedback)
 
-        // Expand schema expressions
+        // Expand all remaining schema expressions
         let expanded = expandSchema(schema.schema, scope, '', false, true, feedback)
 
-        // Write final schema
-        let body = stringify(expanded, (key: any, val: any) => (key === '$templates' || key === '$requires' || key === '$templateDirs' || key === '$examples') ? undefined : val)
-        await generateFile(ppath.join(outPath, `${prefix}.json`), body, force, feedback)
+        if (!error) {
+            // Write final schema
+            let body = stringify(expanded, (key: any, val: any) => (key === '$templates' || key === '$requires' || key === '$templateDirs' || key === '$examples') ? undefined : val)
+            await generateFile(ppath.join(outPath, `${prefix}.json`), body, force, feedback)
 
-        // Merge together all dialog files
-        if (singleton) {
-            if (!merge) {
-                feedback(FeedbackType.info, 'Combining into singleton .dialog')
-                await generateSingleton(scope.prefix, outPath, outDir, feedback)
-            } else {
-                await generateSingleton(scope.prefix, outPath, outPathSingle, feedback)
+            // Merge together all dialog files
+            if (singleton) {
+                if (!merge) {
+                    feedback(FeedbackType.info, 'Combining into singleton .dialog')
+                    await generateSingleton(scope.prefix, outPath, outDir, feedback)
+                } else {
+                    await generateSingleton(scope.prefix, outPath, outPathSingle, feedback)
+                }
             }
-        }
 
-        // Merge old and new directories
-        if (merge) {
-            if (singleton) {
-                await merger.mergeAssets(prefix, outDir, outPathSingle, outDir, allLocales, feedback)
-            } else {
-                await merger.mergeAssets(prefix, outDir, outPath, outDir, allLocales, feedback)
+            // Merge old and new directories
+            if (merge) {
+                if (singleton) {
+                    await merger.mergeAssets(prefix, outDir, outPathSingle, outDir, allLocales, feedback)
+                } else {
+                    await merger.mergeAssets(prefix, outDir, outPath, outDir, allLocales, feedback)
+                }
             }
         }
     } catch (e) {
         feedback(FeedbackType.error, e.message)
     }
+
+    if (error) {
+        externalFeedback(FeedbackType.error, '*** Errors prevented generation ***')
```

Comment:
Consider including the error information in the feedback call

Reply:
It will have already been output.

---
In reply to: [491185296](https://github.com/microsoft/BotBuilder-Samples/pull/2750#discussion_r491185296) [](ancestors = 491185296)